pyproject.toml

```diff
@@ -102,8 +102,12 @@ exclude = [
 ]
 
 [tool.ruff]
-extend-select = ["B0", "C4", "I", "ICN", "ISC", "N", "RUF", "SIM", "UP"]
-extend-ignore = ["ISC001", "RUF005", "RUF012"]
+extend-select = ["B", "C4", "COM", "FA", "FURB", "I", "ICN", "ISC", "N", "PERF", "PGH", "PIE", "RUF", "SIM", "UP"]
+extend-ignore = ["ISC001", "RUF005", "RUF012", "COM812"]
+
+[tool.ruff.per-file-ignores]
+"myst_parser/parsers/docutils_.py" = ["FA"]
+"myst_parser/config/main.py" = ["FA"]
```

Comment:
there is logic in these files currently utilises runtime analysis of types, so you can't have postponed evaluation